services/ui_backend_service/api/dag.py

```diff
@@ -85,6 +72,68 @@ async def get_run_dag(self, request):
 
         return web_response(status, body)
 
+    async def get_run_ids_from_codepackage_metadata(self, request):
+        """
+        Tries to locate 'code-package' or 'code-package-url' in run metadata.
+
+        Returns
+        -------
+        Tuple
+            flow_id : str
+            run_id : str or int
```

Comment:
nit: it doesn't really return a tuple, right?

Reply:
cleaned up now.